src/mjolnir/pbfgraphparser.cc

```diff
@@ -1415,17 +1416,47 @@ struct graph_callback : public OSMPBF::Callback {
       } else {
         results = empty_node_results_;
       }
+      bool is_bss = false;
```

Comment:
i think we can simplify this whole thing by just bailing out early if we dont have the bike rental value:

```c++
    // Handle bike share stations separately
    boost::optional<Tags> results = boost::none;
    if (bss_nodes_) {
      // Get tags - do't bother with Lua callout if the taglist is empty
      if (tags.size() > 0) {
        results = lua_.Transform(OSMType::kNode, osmid, tags);
      } else {
        results = empty_node_results_;
      }
      
      // bail if there is nothing bike related
      Tags::const_iterator found;
      if (!results.has_value() || (found = results->find("amenity")) == results->end() || found->second != "bicycle_rental") {
        return;
      }

      // Create a new node and set its attributes
      OSMNode n{osmid};
      n.set_latlng(lng, lat);
      n.set_type(NodeType::kBikeShare);
      valhalla::BikeShareStationInfo bss_info;

      for (auto& key_value : *results) {
        if (key_value.first == "name") {
          bss_info.set_name(key_value.second);
        } else if (key_value.first == "network") {
          bss_info.set_network(key_value.second);
        } else if (key_value.first == "ref") {
          bss_info.set_ref(key_value.second);
        } else if (key_value.first == "capacity") {
          auto capacity = std::strtoul(key_value.second.c_str(), nullptr, 10);
          if (capacity > 0) {
            bss_info.set_capacity(capacity);
          }
        } else if (key_value.first == "operator") {
          bss_info.set_operator_(key_value.second);
        }
      }

      std::string buffer;
      bss_info.SerializeToString(&buffer);
      n.set_bss_info_index(osmdata_.node_names.index(buffer));
      ++osmdata_.node_name_count;

      bss_nodes_->push_back(n);
      return; // we are done because we are just handling bss this parsing pass
    }
```

Reply:
Done ! :)